project1/server/agent.cpp

```diff
@@ -185,50 +180,19 @@ namespace server {
         }
 
         // monitor for termination request for get and put commands
-        uint16_t totalBytesSent = 0;
-
-        // make 1000 byte buffer
-        int numBytesToSend = 1000;
-        uint8_t buf[numBytesToSend];
-
-        // counter used to track
-        int byte_counter = 0;
-
+        uint16_t kBytesSent = 0;
+        const uint32_t kBytesToSend = 1000;
         bool terminated = false;
 
         // continue until we've sent the entire message
-        while (totalBytesSent < outgoing_message_buffer_.size()) {
-
-
-            // fill the buffer one byte at a time
-            for (int n:outgoing_message_buffer_) {
-
-                // check if 1000 byte buffer is filled
-                if (byte_counter == 1000) {
-                    terminated = !active_commands_->Contains(command_id);
-
-                    // check if the process has been terminated
-                    if (!terminated) {
-                        // Send 1000 bytes of the message.
-                        if (send(client_fd_, buf,
-                                 numBytesToSend, 0) < 0) {
-                            perror("Failed to send the file contents.");
-                            return false;
-                        }
-                    } else {
-                        return false;
-                    }
-
-                    // reset
-                    byte_counter = 0;
-
-                    // update
-                    totalBytesSent += 1000;
-                }
-                buf[byte_counter] = outgoing_message_buffer_.at(n);
-                ++byte_counter;
-
+        while (kBytesSent < outgoing_message_buffer_.size()) {
+            if (!terminated) {
+                kBytesSent += send(client_fd_, outgoing_message_buffer_.data() + kBytesSent,
```

Comment:
There is a bug here if `outgoing_message_buffer_.size()` is not a perfect multiple of 1000, in which case it will try to read past the end of the buffer on the last message. I think in the code snippet I posted about this, I had some logic using `std::min` to handle this case.